src/platform_impl/windows/dark_mode.rs

```diff
@@ -160,40 +165,37 @@ pub fn allow_dark_mode_for_window(hwnd: HWND, is_dark_mode: bool) {
   }
 }
 
-type SetWindowCompositionAttribute =
-  unsafe extern "system" fn(HWND, *mut WINDOWCOMPOSITIONATTRIBDATA) -> BOOL;
-static SET_WINDOW_COMPOSITION_ATTRIBUTE: Lazy<Option<SetWindowCompositionAttribute>> =
-  Lazy::new(|| get_function!("user32.dll", SetWindowCompositionAttribute));
-
-type WINDOWCOMPOSITIONATTRIB = u32;
-const WCA_USEDARKMODECOLORS: WINDOWCOMPOSITIONATTRIB = 26;
-#[repr(C)]
-struct WINDOWCOMPOSITIONATTRIBDATA {
-  Attrib: WINDOWCOMPOSITIONATTRIB,
-  pvData: *mut c_void,
-  cbData: usize,
-}
-
-fn refresh_titlebar_theme_color(hwnd: HWND, is_dark_mode: bool) {
-  // SetWindowCompositionAttribute needs a bigbool (i32), not bool.
-  let mut is_dark_mode_bigbool: i32 = is_dark_mode.into();
-
+fn refresh_titlebar_theme_color(hwnd: HWND, is_dark_mode: bool, redraw_title_bar: bool) {
   if let Some(ver) = *WIN10_BUILD_VERSION {
-    if ver < 18362 {
+    if ver < 17763 {
+      let mut is_dark_mode_bigbool: i32 = is_dark_mode.into();
       unsafe {
         let _ = SetPropW(
           hwnd,
           w!("UseImmersiveDarkModeColors"),
           HANDLE(&mut is_dark_mode_bigbool as *mut _ as _),
         );
       }
-    } else if let Some(set_window_composition_attribute) = *SET_WINDOW_COMPOSITION_ATTRIBUTE {
-      let mut data = WINDOWCOMPOSITIONATTRIBDATA {
-        Attrib: WCA_USEDARKMODECOLORS,
-        pvData: &mut is_dark_mode_bigbool as *mut _ as _,
-        cbData: std::mem::size_of_val(&is_dark_mode_bigbool) as _,
+    } else {
+      // https://github.com/MicrosoftDocs/sdk-api/pull/966/files
+      let dwmwa_use_immersive_dark_mode = if ver > 18985 {
+        DWMWINDOWATTRIBUTE(20)
+      } else {
+        DWMWINDOWATTRIBUTE(19)
       };
-      let _ = unsafe { set_window_composition_attribute(hwnd, &mut data as *mut _) };
+      let dark_mode = BOOL::from(is_dark_mode);
+      unsafe {
+        let _ = DwmSetWindowAttribute(
+          hwnd,
+          dwmwa_use_immersive_dark_mode,
+          &dark_mode as *const BOOL as *const c_void,
+          std::mem::size_of::<BOOL>() as u32,
+        );
+      }
+      if redraw_title_bar {
```

Comment:
why not always re-render titelbar?

Reply:
It's not needed on initial load

Reply:
does it have any side effect if always redraw then? it would just make the code simpler

Reply:
Slightly worth performance, I would like to keep it this way, but if think cleaner code is more important I wouldn't against it

Reply:
nah it is fine.